Lib/posixpath.py

```diff
@@ -500,7 +500,7 @@ def commonpath(paths):
         split_paths = [path.split(sep) for path in paths]
 
         try:
-            isabs, = set(p[:1] == sep for p in paths)
+            isabs, = {p[:1] == sep for p in paths}
```

Comment:
This code smells bad.  Is there a better way to express what it is trying to do?